blog/_src/posts/2022-06-21-shallow-typed-racket.scrbl

```diff
@@ -0,0 +1,367 @@
+    This means that Deep--Shallow and Deep--Optional interactions can be expensive.
+  }
+  @item{
+    Shallow types guard the boundaries to Optional and Untyped code with shape checks.
+  }
+  @item{
+    Optional types never enforce themselves.
+    But a Deep-typed or Shallow-typed client of Optional code will insert
+    run-time checks, as a consequence of their strategies.
+  }
+]
+
+These checks may come as a surprise, especially because all typed code gets validated
+by the same static type checker.
+The trouble is that run-time interactions can mix untyped values into
+some of these typed modules.
+If an Optional module were to import an untyped function and send it to Deep-typed code
+without a contract wrapper, it could break the Deep type invariants.
+
+
+@section{Reflections on Deep, Shallow, and Optional}
+
+Deep types, Shallow types, and Optional types have complementary strengths.
+When and where does each one work best?
+Here are a few suggestions, based on
+@secref["When_to_Use_Deep__Shallow__or_Optional_"
+        #:doc '(lib "typed-racket/scribblings/ts-guide.scrbl")]
+from the Typed Racket Guide:
+
+@itemlist[
+  @item{
+    Deep types make the most of static checking and optimizations.
+    Use them for self-sufficient groups of typed modules.
+    Avoid them at high-traffic boundaries to untyped or non-Deep code.
+  }
+  @item{
+    Shallow types provide a weak but useful soundness guarantee at low cost.
+    Use them in typed modules that frequently communicate with the untyped world.
+    Avoid them, however, in large typed modules because every expression in typed
+    code potentially needs a Shallow shape check.
+  }
+  @item{
+    Optional types use types for static analysis and nothing more.
+    Use them when you do not want types enforced at run-time.
+  }
+]
+
+Overall, we are very excited to be adding these languages to the
+Typed Racket family.
+Learning more about where they fit well in practical applications
+and about how developers tend to use them is part of the adventure.
+
+
+@section[#:tag "sec:further-reading"]{Further Reading}
+
+@itemlist[
+  @item{
+     For the TR basics:
+     @secref["typed-untyped-interaction"
+         #:doc '(lib "typed-racket/scribblings/ts-guide.scrbl")]
+     in the Typed Racket Guide
+  }
+  @item{
+    For more TR details:
+    @secref["behavior-of-types"
+         #:doc '(lib "typed-racket/scribblings/ts-reference.scrbl")]
+    in the Typed Racket Reference
+  }
+  @item{
+    For the theoretical and practical motivation:
+    @linebreak[]
+    Ben Greenman.
+    @italic{Deep and Shallow Types for Gradual Languages}.
+    PLDI 2022.
+    @linebreak[]
+    Paper at @url{https://www2.ccs.neu.edu/racket/pubs/g-pldi-2022.pdf}
+    @linebreak[]
+    Slides at @url{https://cs.brown.edu/people/bgreenma/publications/apples-to-apples/g-pldi-2022-slides.pdf}
+  }
+  @item{
+    Shallow TR is based on the Transient semantics of Reticulated Python,
```

Comment:
I think this should cite the paper.

Reply:
Fixed. I kept the dissertation bullet too, right below.